packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js

```diff
@@ -156,60 +98,72 @@ export const useShippingMethod = props => {
-        }
-    }, [cartId, fetchShippingMethods]);
-
-    useEffect(() => {
-        if (cartId) {
-            fetchSelectedShippingMethod({
+            fetchShippingMethodInfo({
                 variables: { cartId }
             });
         }
-    }, [cartId, fetchSelectedShippingMethod]);
-
-    useEffect(() => {
-        try {
-            const chosenShippingMethod =
-                chosenShippingMethodData.cart.shipping_addresses[0]
-                    .selected_shipping_method;
-
-            const nextDisplayState = chosenShippingMethod
-                ? displayStates.DONE
-                : displayStates.EDITING;
-
-            setDisplayState(nextDisplayState);
-        } catch {
-            setDisplayState(displayStates.EDITING);
-        }
-    }, [chosenShippingMethodData]);
-
-    // Simple heuristic to check shipping data existed prior to this render.
-    // On first submission, when we have data, we should tell the checkout page
-    // so that we set the next step correctly.
-    const hasData =
-        chosenShippingMethodData &&
-        chosenShippingMethodData.cart.shipping_addresses.length &&
-        chosenShippingMethodData.cart.shipping_addresses[0]
-            .selected_shipping_method;
+    }, [cartId, fetchShippingMethodInfo]);
 
+    // When we have data we should tell the checkout page
+    // so that it can set the step correctly.
     useEffect(() => {
         if (hasData) {
             onSave();
         }
     }, [hasData, onSave]);
 
+    useEffect(() => {
+        if (!data) return;
+
+        // Sorts available shipping methods by price.
+        const byPrice = (a, b) => a.amount.value - b.amount.value;
+
+        // Adds a serialized property to available shipping method objects
+        // so they can be selected in the radio group.
+        const addSerializedProperty = shippingMethod => {
+            const serializedValue = serializeShippingMethod(shippingMethod);
+
+            return {
+                ...shippingMethod,
+                serializedValue
+            };
+        };
+
+        // Determine the "primary" shipping address by using
+        // the first shipping address on the cart.
+        const primaryShippingAddress = data.cart.shipping_addresses[0];
+
+        // Shape the list of available shipping methods.
+        // Sort them by price and add a serialized property to each.
+        const rawShippingMethods =
+            primaryShippingAddress.available_shipping_methods;
+        const shippingMethodsByPrice = [...rawShippingMethods].sort(byPrice);
+        const shippingMethods = shippingMethodsByPrice.map(
+            addSerializedProperty
+        );
+        setShippingMethods(shippingMethods);
+
+        // Determine the selected shipping method.
+        // If we don't have one, default to the lowest cost shipping method instead.
+        const selectedMethod = primaryShippingAddress.selected_shipping_method;
+        setSelectedShippingMethod(selectedMethod);
```

Comment:
shippingMethods in are all serialized before setting in state `setShippingMethods(shippingMethods);`.

Is there a reason not to perform serialization for `selectedMethod`?

I ask because if it's done here, we don't need to export and use the `serializeShippingMethod` in `ShippingMethod` component.

Reply:
Resolved by https://github.com/magento/pwa-studio/pull/2382/commits/d43ed986d954ba5e9ce3b1fa07046e9d1fdf0df6.